packages/edit-site/src/components/page-pages/index.js

```diff
@@ -155,6 +161,25 @@ export default function PagePages() {
 
 	const trashPostAction = useTrashPostAction();
 	const actions = useMemo( () => [ trashPostAction ], [ trashPostAction ] );
+	const onChangeView = useCallback(
+		( viewUpdater ) => {
+			let updatedView =
+				typeof viewUpdater === 'function'
+					? viewUpdater( view )
+					: viewUpdater;
+			if ( updatedView.type !== view.type ) {
```

Comment:
Upon view switching, there are view properties _common_ to every view type that we may need to update as well.

- One that is obvious is: the [fields that are hidable](https://github.com/WordPress/gutenberg/pull/55353/#issuecomment-1764499759)
- One that is not obvious is the [fields that are currently hidden](https://github.com/WordPress/gutenberg/pull/55353/#issuecomment-1764525212). We cannot update blindly this field: what if the user has changed the fields visibility? What would happen to the `hiddenFields` provided by the new view type? In this case, I presume we should leave as it is if the user has done modifications. Though, this property could be modified as a side effect of other changes. For example: because `featured-image` is not hidable, we need to make sure it's not in the list of hidden fields.

Reply:
Here's how I would solve it:

 - Add a "requiredFields" config to the "layoutConfig (the sub property that is reset when you switch types)
 - Remove the enableHiding config entirely. 

I'll follow-up with that.

Reply:
Just a couple of nitpicks:

- I'd remove the `config` from the `layout` key – everything is config, the `layout` just happens to be about layout.
- I'd make `layout` be part of the [initial view schema](https://github.com/WordPress/gutenberg/blob/update/page-table-allow-sort-by-date/packages/edit-site/src/components/page-pages/index.js#L28). Even if it's not used by the default `list` view type, that's the only place where we have a "view type schema" of sorts – having it there helps to visualize the current state of things.

Reply:
Renamed the config, and I've thinking about the hiddenFields and enableHiding. I think it's a bit more complex than I expected and the reality is that I'm also not sure about the expectations here. Maybe the "title" should be mandatory for all views for instance. Anyway, let's explore changes here separately.

Reply:
I've added a note about the "required fields" to the overview issue.